source/extensions/filters/http/stateful_session/stateful_session.cc

```diff
@@ -0,0 +1,84 @@
+#include "source/extensions/filters/http/stateful_session/stateful_session.h"
+
+#include <cstdint>
+#include <memory>
+
+#include "source/common/config/utility.h"
+#include "source/common/http/utility.h"
+#include "source/common/upstream/load_balancer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace StatefulSession {
+
+StatefulSessionConfig::StatefulSessionConfig(const ProtoConfig& config,
+                                             Server::Configuration::CommonFactoryContext& context) {
+  if (config.host_statuses_size() > 0) {
+    std::vector<envoy::config::core::v3::HealthStatus> host_statuses;
+    host_statuses.reserve(config.host_statuses_size());
+    for (int i = 0; i < config.host_statuses_size(); i++) {
```

Comment:
prefer range-based loop.

Reply:
get it.

Reply:
I remembered why I wrote the code as this, because `host_statuses()` returns an array of type int instead of an array of type envoy::config::core::v3::HealthStatus, which loses the enum type information.

Reply:
Cool. Comments on this will be very helpful for future readers.